src/internal_telemetry/allocations/allocator/tracing_allocator.rs

```diff
@@ -99,13 +135,13 @@ unsafe impl<A: GlobalAlloc, T: Tracer> GlobalAlloc for GroupedTraceableAllocator
 
 #[inline(always)]
 fn get_wrapped_layout(object_layout: Layout) -> (Layout, usize) {
-    static HEADER_LAYOUT: Layout = Layout::new::<usize>();
+    static HEADER_LAYOUT: Layout = Layout::new::<u8>();
 
     // We generate a new allocation layout that gives us a location to store the active allocation group ID ahead
     // of the requested allocation, which lets us always attempt to retrieve it on the deallocation path.
-    let (actual_layout, offset_to_object) = HEADER_LAYOUT
-        .extend(object_layout)
+    let (actual_layout, offset_to_group_id) = object_layout
+        .extend(HEADER_LAYOUT)
         .expect("wrapping requested layout resulted in overflow");
 
-    (actual_layout, offset_to_object)
+    (actual_layout.pad_to_align(), offset_to_group_id)
```

Comment:
Did you actually need to pad the alignment to get something working?

Reply:
The ```Layout::extend``` docs recommended padding it to match C repr.